### ISSUE TITLE
Failing rpec tests

### ISSUE THREAD

**Issue by bbaja42** [danlucraft/rak#14]
Hi,
I've tried rspec on rak_spec.rb, and following tests fail

Failures:

  1) Rak with no options prints all matches from files in subdirectories
     Failure/Error: rak("Pikon").should == <<-END
       expected: "      _dir1/bar.rb_\n         2|bar bar bar bar _Pikon_ bar\n         9|bar bar _Pikon_ bar bar bar\n\n      _foo.rb_\n         6|foo foo foo foo foo _Pikon_ foo foo\n         8|foo _Pikon_ foo foo foo foo foo foo\n"
            got: "      _foo.rb_\n         6|foo foo foo foo foo _Pikon_ foo foo\n         8|foo _Pikon_ foo foo foo foo foo foo\n\n      _dir1/bar.rb_\n         2|bar bar bar bar _Pikon_ bar\n         9|bar bar _Pikon_ bar bar bar\n" (using ==)
       Diff:

```
   @@ -1,8 +1,8 @@
   -      *dir1/bar.rb*
   -         2|bar bar bar bar *Pikon* bar
   -         9|bar bar *Pikon* bar bar bar
   -
          *foo.rb*
             6|foo foo foo foo foo *Pikon* foo foo
             8|foo *Pikon* foo foo foo foo foo foo
   +
   +      *dir1/bar.rb*
   +         2|bar bar bar bar *Pikon* bar
   +         9|bar bar *Pikon* bar bar bar
 # ./spec/rak_spec.rb:27
```

  2) Rak options -h suppresses filename and line number printing
     Failure/Error: rak("Pik -h").should == <<-END
       expected: "      bar bar bar bar _Pik_on bar\n      bar bar _Pik_on bar bar bar\n      foo foo foo foo foo _Pik_on foo foo\n      foo _Pik_on foo foo foo foo foo foo\n"
            got: "      foo foo foo foo foo _Pik_on foo foo\n      foo _Pik_on foo foo foo foo foo foo\n      bar bar bar bar _Pik_on bar\n      bar bar _Pik_on bar bar bar\n" (using ==)
       Diff:

```
   @@ -1,5 +1,5 @@
   -      bar bar bar bar *Pik*on bar
   -      bar bar *Pik*on bar bar bar
          foo foo foo foo foo *Pik*on foo foo
          foo *Pik*on foo foo foo foo foo foo
   +      bar bar bar bar *Pik*on bar
   +      bar bar *Pik*on bar bar bar
 # ./spec/rak_spec.rb:168
```

  3) Rak options --follow means follow symlinks
     Failure/Error: rak("Sagitarron --follow", :ansi => '').should == <<-END
       expected: "      corge.rb\n         1|corge corge corge Sagitarron corge\n\n      ln_dir/corge.rb\n         1|corge corge corge Sagitarron corge\n"
            got: "      ln_dir/corge.rb\n         1|corge corge corge Sagitarron corge\n\n      corge.rb\n         1|corge corge corge Sagitarron corge\n" (using ==)
       Diff:

```
   @@ -1,6 +1,6 @@
   -      corge.rb
   +      ln_dir/corge.rb
             1|corge corge corge Sagitarron corge

   -      ln_dir/corge.rb
   +      corge.rb
             1|corge corge corge Sagitarron corge
 # ./spec/rak_spec.rb:356
```

  4) Rak with combinations of options -h and redirection
     Failure/Error: rak("Pik -h | cat", :test_mode => false).should == <<-END
       expected: "      bar bar bar bar Pikon bar\n      bar bar Pikon bar bar bar\n      foo foo foo foo foo Pikon foo foo\n      foo Pikon foo foo foo foo foo foo\n"
            got: "      foo foo foo foo foo Pikon foo foo\n      foo Pikon foo foo foo foo foo foo\n      bar bar bar bar Pikon bar\n      bar bar Pikon bar bar bar\n" (using ==)
       Diff:

```
   @@ -1,5 +1,5 @@
   -      bar bar bar bar Pikon bar
   -      bar bar Pikon bar bar bar
          foo foo foo foo foo Pikon foo foo
          foo Pikon foo foo foo foo foo foo
   +      bar bar bar bar Pikon bar
   +      bar bar Pikon bar bar bar
 # ./spec/rak_spec.rb:491
```

  5) Rak with --eval should support break
     Failure/Error: rak(%Q[--eval 'break if $. == 2']).should == <<-END
       expected: "      _dir1/bar.rb_\n         1|_\n\n      *foo.rb_\n         1|_\n\n      *quux.py_\n         1|_quux quux quux quux Virgon quux quux_\n\n      _Rakefile_\n         1|_rakefile rakefile Canceron rakefile_\n\n      _shebang_\n         1|_#!/usr/bin/env ruby_\n"
            got: "      _foo.rb_\n         1|_\n\n      *quux.py_\n         1|_quux quux quux quux Virgon quux quux_\n\n      _shebang_\n         1|_#!/usr/bin/env ruby_\n\n      _Rakefile_\n         1|_rakefile rakefile Canceron rakefile_\n\n      _dir1/bar.rb_\n         1|*\n" (using ==)
       Diff:

```
   @@ -1,15 +1,15 @@
   -      *dir1/bar.rb*
   -         1|*
   -
          *foo.rb*
             1|*

          *quux.py*
             1|*quux quux quux quux Virgon quux quux*

   +      *shebang*
   +         1|*#!/usr/bin/env ruby*
   +
          *Rakefile*
             1|*rakefile rakefile Canceron rakefile*

   -      *shebang*
   -         1|*#!/usr/bin/env ruby*
   +      *dir1/bar.rb*
   +         1|*
 # ./spec/rak_spec.rb:509
```

  6) Rak with --eval should support next and contexts
     Failure/Error: rak(%Q[-C 2 --eval 'next unless $_ =~ /Pikon/']).should == <<-END
       expected: "      _dir1/bar.rb_\n         1|\n         2|bar bar bar bar _Pikon_ bar\n         3| \n         4|\n         7|\n         8|\n         9|bar bar _Pikon_ bar bar bar\n\n      _foo.rb_\n         4|foo Capsicum foo foo foo foo foo\n         5|\n         6|foo foo foo foo foo _Pikon_ foo foo\n         7|\n         8|foo _Pikon_ foo foo foo foo foo foo\n         9|\n        10|foo foo Six foo foo foo Six foo\n"
            got: "      _foo.rb_\n         4|foo Capsicum foo foo foo foo foo\n         5|\n         6|foo foo foo foo foo _Pikon_ foo foo\n         7|\n         8|foo _Pikon_ foo foo foo foo foo foo\n         9|\n        10|foo foo Six foo foo foo Six foo\n\n      _dir1/bar.rb_\n         1|\n         2|bar bar bar bar _Pikon_ bar\n         3| \n         4|\n         7|\n         8|\n         9|bar bar _Pikon_ bar bar bar\n" (using ==)
       Diff:
       @@ -1,12 +1,3 @@
       -      _dir1/bar.rb_
       -         1|
       -         2|bar bar bar bar _Pikon_ bar
       -         3| 
       -         4|
       -         7|
       -         8|
       -         9|bar bar _Pikon_ bar bar bar
       -
              _foo.rb_
                 4|foo Capsicum foo foo foo foo foo
                 5|
       @@ -15,4 +6,13 @@
                 8|foo _Pikon_ foo foo foo foo foo foo
                 9|
                10|foo foo Six foo foo foo Six foo
       +
       +      _dir1/bar.rb_
       +         1|
       +         2|bar bar bar bar _Pikon_ bar
       +         3| 
       +         4|
       +         7|
       +         8|
       +         9|bar bar _Pikon_ bar bar bar
     # ./spec/rak_spec.rb:544

  7) Rak with --eval should support break and contexts (and matches past break should get no highlighting)
     Failure/Error: rak(%Q[-C 2 --eval 'next unless $_[/Pikon/]...(break; nil)']).should == <<-END
       expected: "      _dir1/bar.rb_\n         1|\n         2|bar bar bar bar _Pikon_ bar\n         3| \n         4|\n\n      _foo.rb_\n         4|foo Capsicum foo foo foo foo foo\n         5|\n         6|foo foo foo foo foo _Pikon_ foo foo\n         7|\n         8|foo Pikon foo foo foo foo foo foo\n"
            got: "      _foo.rb_\n         4|foo Capsicum foo foo foo foo foo\n         5|\n         6|foo foo foo foo foo _Pikon_ foo foo\n         7|\n         8|foo Pikon foo foo foo foo foo foo\n\n      _dir1/bar.rb_\n         1|\n         2|bar bar bar bar _Pikon_ bar\n         3| \n         4|\n" (using ==)
       Diff:

```
   @@ -1,13 +1,13 @@
   -      *dir1/bar.rb*
   -         1|
   -         2|bar bar bar bar *Pikon* bar
   -         3| 
   -         4|
   -
          *foo.rb*
             4|foo Capsicum foo foo foo foo foo
             5|
             6|foo foo foo foo foo *Pikon* foo foo
             7|
             8|foo Pikon foo foo foo foo foo foo
   +
   +      *dir1/bar.rb*
   +         1|
   +         2|bar bar bar bar *Pikon* bar
   +         3| 
   +         4|
 # ./spec/rak_spec.rb:566
```

  8) Rak with --eval should support multiple matches
     Failure/Error: rak(%Q[--eval 'next unless $_ =~ /Pikon/; $_.scan(/\b\w{3}\b/){ matches << $~ }']).should == <<-END
       expected: "      _dir1/bar.rb_\n         2|_bar_ _bar_ _bar_ _bar_ Pikon _bar_\n         9|_bar_ _bar_ Pikon _bar_ _bar_ _bar_\n\n      _foo.rb_\n         6|_foo_ _foo_ _foo_ _foo_ _foo_ Pikon _foo_ _foo_\n         8|_foo_ Pikon _foo_ _foo_ _foo_ _foo_ _foo_ _foo_\n"
            got: "      _foo.rb_\n         6|_foo_ _foo_ _foo_ _foo_ _foo_ Pikon _foo_ _foo_\n         8|_foo_ Pikon _foo_ _foo_ _foo_ _foo_ _foo_ _foo_\n\n      _dir1/bar.rb_\n         2|_bar_ _bar_ _bar_ _bar_ Pikon _bar_\n         9|_bar_ _bar_ Pikon _bar_ _bar_ _bar_\n" (using ==)
       Diff:

```
   @@ -1,8 +1,8 @@
   -      *dir1/bar.rb*
   -         2|*bar* *bar* *bar* *bar* Pikon *bar*
   -         9|*bar* *bar* Pikon *bar* *bar* *bar*
   -
          *foo.rb*
             6|*foo* *foo* *foo* *foo* *foo* Pikon *foo* *foo*
             8|*foo* Pikon *foo* *foo* *foo* *foo* *foo* *foo*
   +
   +      *dir1/bar.rb*
   +         2|*bar* *bar* *bar* *bar* Pikon *bar*
   +         9|*bar* *bar* Pikon *bar* *bar* *bar*
 # ./spec/rak_spec.rb:596
```

I've tried on both ruby 1.8.7 and 1.9.3, OS is CentOs 6. If you need more info, feel free to let me know.

**Comment by bbaja42** [comment]
uname -a
Linux localhost.localdomain 2.6.32-220.7.1.el6.x86_64 #1 SMP Wed Mar 7 00:52:02 GMT 2012 x86_64 x86_64 x86_64 GNU/Linux

**Comment by stefanor** [comment]
Fixed in #17
